compiler/plc_xml/src/model/pou.rs

```diff
@@ -39,6 +44,41 @@ impl<'xml> Pou<'xml> {
     }
 }
 
+impl<'xml> Parseable for Pou<'xml> {
+    fn visit(reader: &mut Reader, tag: Option<BytesStart>) -> Result<Self, Error> {
+        let Some(tag) = tag else {
+            unreachable!()
+        };
+        let attributes = get_attributes(tag.attributes())?;
+        let mut pou = Pou::default().with_attributes(attributes)?;
+        loop {
+            match reader.read_event()? {
+                Event::Start(tag) if tag.name().as_ref() == b"interface" => {
+                    // XXX: this is very specific to our own xml schema, but does not adhere to the plc open standard
+                    pou.interface =
+                        Some(Interface::visit(reader, Some(tag))?.append_end_keyword(&pou.pou_type));
+                }
+                Event::Start(tag) if tag.name().as_ref() == b"body" => {
+                    pou.body = Body::visit(reader, Some(tag))?;
+                }
+                Event::Start(tag) if tag.name().as_ref() == b"actions" => {
+                    let actions: Vec<Action<'_>> = Parseable::visit(reader, Some(tag))?;
+                    for mut action in actions.into_iter() {
+                        // Copy the action type names
+                        action.type_name = pou.name.clone();
+                        pou.actions.push(action);
+                    }
+                }
```

Comment:
While this is almost certainly a matter of taste, I do feel like this is more readable:

```rust
match reader.read_event()? {
    Event::Start(tag) => {
        match tag.name().as_ref() {
            // XXX: this is very specific to our own xml schema, but does not adhere to the plc open standard
            b"interface" => pou.interface =
                Some(Interface::visit(reader, Some(tag))?.append_end_keyword(&pou.pou_type));
            b"body" => pou.body = Body::visit(reader, Some(tag))?;
            b"actions" => {
                let actions: Vec<Action<'_>> = Parseable::visit(reader, Some(tag))?;
                for mut action in actions.into_iter() {
                    // Copy the action type names
                    action.type_name = pou.name.clone();
                    pou.actions.push(action);
                }
            }
        }
        ...
    }
}
```

Reply:
this is the formatter's doing not mine :)